### PR TITLE
Allow the AWS backend to get credentials from its environment.

### PR DESCRIPTION
**Pull request by tomwilkie**
If you give the VMs the right instance profile/role, they can authenticate to DynamoDB without having credentials on the command line.

Signed-off-by: Tom Wilkie <tom.wilkie@gmail.com>

### CODE REVIEW

**Comment by tomwilkie** [comment]
Tested, works like a charm.

**Comment by tomwilkie** [comment]
Had to fiddle with dep, and upgrade to 0.5.0, as client_golang, weaveworks/common and uber/jaeger-lib were conflicting.

**Comment by tomwilkie** [comment]
@bboreham PTAL

File: Gopkg.toml [hunk 1]
```diff
@@ -6,6 +6,21 @@
   name = "github.com/aws/aws-sdk-go"
   version = "v1.10.8"
 
+[[constraint]]
```
Comment (bboreham) [inline Gopkg.toml]: Why?  I would expect master to be the default in a repo with no releases.

Comment (tomwilkie) [reply Gopkg.toml]: > Had to fiddle with dep, and upgrade to 0.5.0, as client_golang, weaveworks/common and uber/jaeger-lib were conflicting.

Comment (bboreham) [reply Gopkg.toml]: What I need here, in the file, is the reason for the constraint, and the conditions under which it can be removed.

**Comment by bboreham** [comment]
Please update the PR title and description to include all the material changes.

**Comment by bboreham** [comment]
Since I want to update common too, I rebased and tweaked this.

The changes are:
 - update Gopkg.lock to the format of `dep` v0.5.0
 - Don't use full URL as metric label on unmatched path (weaveworks/common#108)
 - Raise default server timeouts from 5s to 30s (weaveworks/common#109)
 - Return errors when http/grpc servers exit. (weaveworks/common#92)
 - Expose jaeger metrics (weaveworks/common#114)
 - Allow the AWS backend to get credentials from its environment. (weaveworks/common#111)

**Comment by bboreham** [comment]
Hmm, I thought GitHub would pick up my branch in this PR - https://github.com/weaveworks/cortex/compare/master...aws-sts

**Comment by bboreham** [comment]
@tomwilkie let me know if you'd like me to push the rebase to your branch

**Comment by tomwilkie** [comment]
I've updated the vendored Prometheus to bring in https://github.com/prometheus/prometheus/pull/4436, better tracing spans in PromQL.

**Pull request merged**